lib/util/import-target.js

```diff
@@ -309,6 +310,10 @@ module.exports = class ImportTarget {
             this.resolverConfig.extensions = this.options.extensions
         }
 
+        if (this.options.modules) {
+            this.resolverConfig.modules = this.options.modules
+        }
```

Comment:
**question(non-blocking)**: I am thinking it may be best to allow all `enhanced-resolve.resolverConfig` through. 
Could the option be `this.options.resolverConfig.modules` instead of `this.options.modules`.

What do you think?

Reply:
It does add some complexity on how that setting would be handled:

- Is it simply the default and are all the settings from this file (`conditionNames`, `mainFiels`, `mainFiles`, `extensions`, ...) applied over the `resolverConfig` option the consumer supplies
- Or is it an override, as in every key the consumer sets on their `resolverConfig` option will override the plugin's values. (I suspect this won't work as the plugin detects some of these based on the file that's currently being linted.)
- Or is it a mix of the two based on which key is being evaluated
- Can you pass a function that gives you the plugin's `resolverConfig` (and possible also some extra context), allowing you to adapt it to your needs (possible on a per-file basis).

If the answer is "definitively only that first option", I agree that adding `resolverConfig` as an option is way better than what I did here.

Reply:
I was thinking that anything specified by the user should overwrite all our configs. eg like this as a final step:

```js
this.resolverConfig = { ...this.resolverConfig, ...this.options.resolverConfig };
```

I think that overwriting the users config with our config would lead to confusion. That also applies to a deep merge.

The function in to mutate the config is also reasonable, but also slightly more complex that I think this warrants -- Not that I am opposed if you like the idea! :sweat_smile:

Reply:
Updated. I omitted the function to mutate, for now. As you mentioned: it adds complexity to the documentation, and no one may ever need it.

Reply:
Updated the PR title and description too for future reference.